src/plugins/expressions/common/expression_functions/specs/kibana.ts

```diff
@@ -44,6 +62,10 @@ export const kibana: ExpressionFunctionKibana = {
 
   args: {},
 
+  migrations: {
+    '7.10.1': (MIGRATE7_10_1 as any) as MigrateFunction,
```

Comment:
this is the part i really don't like. any suggestions ?

Reply:
Why doesn't TS like this? Looks as if it should work without casting to me, but need to test it

Reply:
Maybe part of the issue is that in `expression_function.ts` you are re-defining the migration type instead of reusing `MigrateFunction`? (I assume that's what's being used behind the scenes here?)

Reply:
the problem is that MigrateFunction there is <SerializableSTate,SerializableState> where MIGRATE7_10_1 actually work with well defined types

Reply:
> the problem is that MigrateFunction there is <SerializableSTate,SerializableState> where MIGRATE7_10_1 actually work with well defined types

Okay, after looking into this more, there are actually a couple problems. First, `migrations` isn't in `ExpressionFunctionDefinition` at all, which is masked by the casting.

More problematic is that `ExpressionFunctionDefinition` would need to provide a generic type param so you can specify the actual type(s) of your migrations, that way TS doesn't complain because the functions you provide are too specific.

The bummer is that this makes things more verbose and makes `ExpressionFunctionDefinition` even more complex, but usage would essentially look like:

```ts
interface Migrations extends Record<string, any> {
  '7.10.1': MigrateFunction<MYSTATE_7_10_0, MYSTATE_7_10_1>;
}

export type ExpressionFunctionKibana = ExpressionFunctionDefinition<
  'kibana',
  ExpressionValueSearchContext | null,
  object,
  ExpressionValueSearchContext,
  ExecutionContext,
  Migrations
>;
```

And then `ExpressionFunctionDefinition` would need to take a generic param along the lines of:
```ts
  Migrations extends Record<string, MigrateFunction> = Record<string, MigrateFunction>
```

I tested this and it worked for me locally, but LMK what you think. Either way if we are adding `migrations` to the functions, we should update `ExpressionFunctionDefinition`